buildenv

```diff
@@ -175,49 +160,122 @@ zopen_build() {
-  done
+
+  echo "** APPLY PATCHES, PAX and OVERLAYS BEGIN ***********" 
+  echo "** APPLY PATCHES, PAX and OVERLAYS BEGIN ***********" 
+  echo "** APPLY PATCHES, PAX and OVERLAYS BEGIN ***********" 
+  echo "** APPLY PATCHES BEGIN ***********" 
+  echo "** APPLY PATCHES BEGIN ***********" 
+  echo "** APPLY PATCHES BEGIN ***********" 
+  cd ${ZOPEN_ROOT}/v8base/v8
+  echo "  -- compbuild"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.compbuild.diff
+  echo "  -- cpp"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.cpp.diff
+  echo "  -- debug"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.debug.diff
+  echo "  -- v8"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.diff
+  echo "  -- flags"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.flags.diff
+  echo "  -- killharder"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.killharder.diff
+  echo "  -- memset0"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.memset0.diff
+  echo "  -- teststatus"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.teststatus.diff
+  echo "  -- tls"
+  git apply ${ZOPEN_ROOT}/patches/git.v8.tls.diff
+  echo "Apply patch for third_party repos"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party
+  echo "  -- abseill-cpp"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/abseil-cpp
+  git apply ${ZOPEN_ROOT}/patches/third_party/abseil-cpp/git.diff
+  echo "  -- fuzztest"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/fuzztest/src
+  git apply ${ZOPEN_ROOT}/patches/third_party/fuzztest/src/git.diff
+  echo "  -- googletest"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/googletest/src
+  git apply ${ZOPEN_ROOT}/patches/third_party/googletest/src/git.diff
+  echo "  -- protobuf_chrome"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/protobuf_chrome
+  git apply ${ZOPEN_ROOT}/patches/third_party/protobuf_chrome/git.diff
+  echo "  -- re2"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/re2/src
+  git apply ${ZOPEN_ROOT}/patches/third_party/re2/src/git.diff
+  echo "  -- zlib"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/zlib
+  git apply ${ZOPEN_ROOT}/patches/third_party/zlib/git.diff
+  echo "Apply patches for overall repo complete"
+  echo "Apply patches for overall repo complete"
+  echo "Apply patches for overall repo complete"
+
+  echo "Perform file overlays"
+  echo "Perform file overlays"
+  echo "Perform file overlays"
+  cd ${ZOPEN_ROOT}/v8base/v8/
+  echo "nothing done. clarify.  TODO"
+  echo "  -- patches/buildtools/zos/ar"
+  echo "  -- patches/src/base/debug/stack_trace_zos.cc"
+  echo "  -- patches/src/base/platform/platform-zos.cc"
+  echo "  -- patches/src/heap/src/asm/zos/push_registers_asm.cc"
+  echo "  -- patches/src/snapshot/embedded/platform-embedded-file-writer-zos.cc"
+  echo "  -- patches/src/snapshot/embedded/platform-embedded-file-writer-zos.h"
+  echo "Perform file overlays complete"
+  echo "Perform file overlays complete"
+  echo "Perform file overlays complete"
 
   echo "Apply pax files to buildtools"
-  pushd buildtools
+  echo "Apply pax files to buildtools"
+  echo "Apply pax files to buildtools"
```

Comment:
Why is the echo command repeated (here and elsewhere)?

Reply:
Its repeated to make an easy to read banner when examining the log.

Reply:
But it's unconventional. You may want to instead replace echo by banner in your local repo before running the script.